cmd/collectors/restperf/restperf.go

```diff
@@ -0,0 +1,916 @@
+	}
+
+	err = rest.FetchData(r.Client, href, &records)
+	if err != nil {
+		r.Logger.Error().Stack().Err(err).Str("href", href).Msg("Failed to fetch data")
+		return nil, err
+	}
+
+	all := rest.Pagination{
+		Records:    records,
+		NumRecords: len(records),
+	}
+
+	content, err = json.Marshal(all)
+	if err != nil {
+		r.Logger.Error().Err(err).Str("ApiPath", r.prop.query).Msg("Unable to marshal rest pagination")
+	}
+
+	if !gjson.ValidBytes(content) {
+		return nil, fmt.Errorf("json is not valid for: %s", r.prop.query)
+	}
+
+	results := gjson.GetManyBytes(content, "num_records", "records")
+	numRecords := results[0]
+	if numRecords.Int() == 0 {
+		return nil, errors.New(errors.ERR_NO_INSTANCE, "no "+r.Object+" instances on cluster")
+	}
+
+	r.Logger.Debug().Str("object", r.Object).Str("records", numRecords.String()).Msg("Extracted records")
+
+	results[1].ForEach(func(key, instanceData gjson.Result) bool {
+		var (
+			instanceKey string
+		)
+
+		if !instanceData.IsObject() {
+			r.Logger.Warn().Str("type", instanceData.Type.String()).Msg("Instance data is not object, skipping")
+			return true
+		}
+
+		// extract instance key(s)
+		for _, k := range r.prop.instanceKeys {
+			value := parseProperties(instanceData, k)
+			if value.Exists() {
+				instanceKey += value.String()
+			} else {
+				r.Logger.Warn().Str("key", k).Msg("skip instance, missing key")
+				break
+			}
+		}
+
+		if oldInstances.Delete(instanceKey) {
+			// instance already in cache
+			r.Logger.Debug().Msgf("updated instance [%s%s%s%s]", color.Bold, color.Yellow, key, color.End)
+		} else if _, err := r.Matrix.NewInstance(instanceKey); err != nil {
+			r.Logger.Error().Err(err).Str("Instance key", instanceKey).Msg("add instance")
+		} else {
+			r.Logger.Debug().
+				Str("key", instanceKey).
+				Msg("Added new instance")
+		}
+		return true
+
+	})
+
+	for key := range oldInstances.Iter() {
+		r.Matrix.RemoveInstance(key)
+		r.Logger.Debug().Msgf("removed instance [%s]", key)
+	}
+
+	removed = oldInstances.Size()
+	newSize = len(r.Matrix.GetInstances())
+	added = newSize - (oldSize - removed)
+
+	r.Logger.Debug().Msgf("added %d new, removed %d (total instances %d)", added, removed, newSize)
+
+	if newSize == 0 {
+		return nil, errors.New(errors.ERR_NO_INSTANCE, "")
+	}
+
+	return nil, err
+}
+
+func (r *RestPerf) CollectAutoSupport(p *collector.Payload) {
+	var exporterTypes []string
+	for _, exporter := range r.Exporters {
+		exporterTypes = append(exporterTypes, exporter.GetClass())
+	}
+
+	var counters = make([]string, 0)
+	for k := range r.prop.counters {
+		counters = append(counters, k)
+	}
+
+	var schedules = make([]collector.Schedule, 0)
+	tasks := r.Params.GetChildS("schedule")
+	if tasks != nil && len(tasks.GetChildren()) > 0 {
+		for _, task := range tasks.GetChildren() {
+			schedules = append(schedules, collector.Schedule{
+				Name:     task.GetNameS(),
+				Schedule: task.GetContentS(),
+			})
+		}
+	}
+
+	// Add collector information
```

Comment:
function from rest.go should handle the same.

Reply:
not atm. Both have different props. I'll leave this comment open to see that later.